bundle/View/Redirect/Resolver.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Bundle\EzPlatformSiteApiBundle\View\Redirect;
+
+use Netgen\Bundle\EzPlatformSiteApiBundle\View\ContentView;
+use Netgen\EzPlatformSiteApi\API\Values\Content;
+use Netgen\EzPlatformSiteApi\API\Values\Location;
+use Symfony\Component\Routing\RouterInterface;
+
+final class Resolver
+{
+    /**
+     * @var \Netgen\Bundle\EzPlatformSiteApiBundle\QueryType\ParameterProcessor
+     */
+    private $parameterProcessor;
+
+    /**
+     * @var \Symfony\Component\Routing\RouterInterface
+     */
+    private $router;
+
+    /**
+     * Resolver constructor.
+     *
+     * @param \Netgen\Bundle\EzPlatformSiteApiBundle\View\Redirect\ParameterProcessor $parameterProcessor
+     * @param \Symfony\Component\Routing\RouterInterface $router
+     */
+    public function __construct(
+        ParameterProcessor $parameterProcessor,
+        RouterInterface $router
+    ) {
+        $this->parameterProcessor = $parameterProcessor;
+        $this->router = $router;
+    }
+
+    /**
+     * Builds a path to the redirect target.
+     *
+     * Value from configuration can be:
+     *      - SiteAPI location
+     *      - SiteAPI content
+     *      - location id fetched from config resolver
+     *
+     * @param string $redirectConfig
+     * @param \Netgen\Bundle\EzPlatformSiteApiBundle\View\ContentView $view
+     *
+     * @return string
+     */
+    public function resolveTarget(string $redirectConfig, ContentView $view): string
+    {
+        $value = $this->parameterProcessor->process($redirectConfig, $view);
+
+        if (is_array($value)) {
```

Comment:
In what cases is this value an array?

Reply:
for example, `@=location.getChildren(1)`

Reply:
Can't that be written as `@=location.getChildren(1)[0]` ? It seems weird supporting arrays for such a trivial usecase, especially when you only use the first value. Kinda implies that if you use only a first value of an array, you can just as well transfer the value itself.

Reply:
Damn, that works. Will remove!